application/src/main/java/org/togetherjava/tjbot/commands/help/LeaversPurgeDatabaseRoutine.java

```diff
@@ -0,0 +1,46 @@
+package org.togetherjava.tjbot.commands.help;
+
+import net.dv8tion.jda.api.JDA;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.Routine;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.generated.tables.AddHelpChannel;
+
+import java.time.Instant;
+import java.time.Period;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Purge Routine to get rid of old thread creations in the database.
+ */
+public class LeaversPurgeDatabaseRoutine implements Routine {
+    private final Database database;
+    private static final Logger logger = LoggerFactory.getLogger(LeaversPurgeDatabaseRoutine.class);
+    private static final Period DELETE_MESSAGE_RECORDS_AFTER = Period.ofDays(7);
+
+    public LeaversPurgeDatabaseRoutine(@NotNull Database database) {
+        this.database = database;
+    }
+
+
+    @Override
+    public @NotNull Schedule createSchedule() {
+        return new Schedule(ScheduleMode.FIXED_RATE, 0, 4, TimeUnit.HOURS);
+    }
+
+    @Override
+    public void runRoutine(@NotNull JDA jda) {
+        int recordsDeleted = database
+            .writeAndProvide(content -> content.deleteFrom(AddHelpChannel.ADD_HELP_CHANNEL))
```

Comment:
Maybe it's me, but I really don't understand when it gets deleted from the database.

Would appreciate some comments on when it gets deleted.

Reply:
Well if the database is going to be used for all help_threads related, it would probably make sense to not purge all the help channels and maybe only the ones which are tied to the user leaving.

Reply:
(its okay if we purge our metadata at regular intervals - maybe after a month). we dont want our database to pile up (disk space is limited) and we also dont want to hold too much info, privacy-wise

Reply:
Well realistically, what would make the most sense interval-wise? 7 days might be too short but I also think a month might be too long.

Reply:
its not a big deal if the table lists archived threads though. with the renaming, it will function as general purpose metadata table for help threads - so other functionalities/commands will consume it as well, not just ur auto-close thing.

Reply:
I'm not saying we shouldn't purge, I'm just saying I do not understand this code, might be because I'm reading from GitHub.

Reply:
> I'm not saying we shouldn't purge, I'm just saying I do not understand this code, might be because I'm reading from GitHub.

- Get access to read and write to the database
- delete all from help_threads where the help time it was created at is <= the difference between now and the time the thread was created.